dali/kernels/imgproc/geom/remap.h

```diff
@@ -0,0 +1,112 @@
+ * | REFLECT     | fedcba|abcdefgh|hgfedcb                       |
+ * --------------------------------------------------------------|
+ * | REFLECT_101 | gfedcb|abcdefgh|gfedcba                       |
+ * --------------------------------------------------------------|
+ * | WRAP        | cdefgh|abcdefgh|abcdefg                       |
+ * --------------------------------------------------------------|
+ * | TRANSPARENT | uvwxyz|abcdefgh|ijklmno                       |
+ * --------------------------------------------------------------|
+ * | ISOLATED    | do not look outside of ROI                    |
+ * --------------------------------------------------------------|
+ */
+enum class BorderType {
+  CONSTANT,
+  REPLICATE,
+  REFLECT,
+  REFLECT_101,
+  WRAP,
+  TRANSPARENT,
+  ISOLATED
+};
+
+template <typename T>
+struct Border {
+  BorderType type;
+  std::optional<T> value;
+};
+
+/**
+ * API for Remap operation. Remap applies a generic geometrical transformation to an image.
+ * @tparam Backend
+ * @tparam T
+ */
+template <typename Backend, typename T>
+struct RemapKernel {
+  /**
+   * Perform remap algorithm. This function allows to apply a different transformation to every
+   * sample in a batch. For a special case, where the same transformation is applied for every
+   * sample in a batch, please refer to corresponding overload.
+   *
+   * Every output sample has to have the same shape as the corresponding input sample.
+   *
+   * Handles only HWC layout.
+   *
+   * The transformation is described by `mapx` and `mapy` parameters, where:
+   * output(x,y) = input( mapx(x,y) , mapy(x,y) )
+   *
+   * When a ROI is empty, it is assumed that it covers whole input image.
+   *
+   * @param context Context for the operation.
+   * @param output Output batch.
+   * @param input Input batch.
+   * @param output_rois ROIs of the output images. Shall be the same size as input_rois.
+   * @param input_rois ROIs of the input images.
+   * @param mapsx Arrays of floats, that determine the transformation.
+   * @param mapsy Arrays of floats, that determine the transformation.
+   * @param interpolations Determines, which interpolation shall be used.
+   * @param borders Determines, how to handle pixels on a border on an image (or ROI).
+   */
+  void Run(KernelContext &context, TensorListView<Backend, T> output,
+           TensorListView<Backend, const T> input, span<Box<2, int64_t>> output_rois,
+           span<Box<2, int64_t>> input_rois, span<TensorView<Backend, const float, 2>> mapsx,
+           span<TensorView<Backend, const float, 2>> mapsy, span<DALIInterpType> interpolations,
+           span<Border<T>> borders) = 0;
+
+
+  /**
+   * Convenient overload. This function shall be used when the transformation parameters are the
+   * same for every sample in an input batch.
+   */
+  void Run(KernelContext &context, TensorListView<Backend, T> output, Box<2, int64_t> output_roi,
```

Comment:
That limits the whole kernel to processing only one size. Maybe you can have different inputs and some roi that fits in all of them, but if roi is for the whole image, it can only be the same size do to the maps being repeated.